tests/test_generate_trajectories.py

```diff
@@ -0,0 +1,107 @@
+
+
+@pytest.fixture(
+    params=[
+        pytest.param(
+            (data, model, early_fusion, input_encoder, backbone),
+            id=f"{data}-{model}-earlyfusion{early_fusion}-{input_encoder}-{backbone}",
+        )
+        for data, model, early_fusion, input_encoder, backbone in [
+            ("pytorch_dataset", "eic_forecasting", None, "eic_encoder", "transformer_decoder"),
+            ("multiwindow_pytorch_dataset", "eic_forecasting", None, "eic_encoder", "transformer_decoder"),
+        ]
+    ]
+)
+def get_kwargs(request, meds_dir) -> dict:
+    def helper(extra_overrides: list[str] = []):
+        data, model, early_fusion, input_encoder, backbone = request.param
+        overrides, raises_value_error, supervised = get_overrides_and_exceptions(
+            data, model, early_fusion, input_encoder, backbone
+        )
+        cfg = create_cfg(overrides=overrides + extra_overrides, meds_dir=meds_dir, supervised=supervised)
+        return dict(
+            cfg=cfg,
+            raises_value_error=raises_value_error,
+            input_kwargs=dict(
+                data=data,
+                model=model,
+                early_fusion=early_fusion,
+                input_encoder=input_encoder,
+                backbone=backbone,
+            ),
+        )
+
+    return helper
+
+
+@pytest.mark.slow
+def test_train_predict(tmp_path: Path, get_kwargs, meds_dir) -> None:  # noqa: F811
+    """Tests training and evaluation by training for 1 epoch with `train.py` then evaluating with `eval.py`.
+
+    :param tmp_path: The temporary logging path.
+    :param cfg_train: A DictConfig containing a valid training configuration.
+    :param cfg_eval: A DictConfig containing a valid evaluation configuration.
+    """
+    kwargs = get_kwargs()
+    cfg_train = kwargs["cfg"]
+    raises_value_error = kwargs["raises_value_error"]
+    with open_dict(cfg_train):
+        cfg_train.trainer.max_epochs = 1
+        cfg_train.test = True
+        cfg_train.paths.output_dir = str(tmp_path)
+        if "multiwindow_pytorch_dataset" == cfg_train.data.name:
+            cfg_train.data.default_window_name = "pre"
+    HydraConfig().set_config(cfg_train)
+    if raises_value_error:
+        with pytest.raises(hydra.errors.InstantiationException):
+            train_main(cfg_train)
+    else:
+        train_main(cfg_train)
+        time_output_dir = Path(
+            subprocess.run(
+                ["meds-torch-latest-dir", f"path={tmp_path}"], capture_output=True, text=True
+            ).stdout.strip()
+        )
+        assert "last.ckpt" in os.listdir(time_output_dir / "checkpoints")
+        overrides, _, supervised = get_overrides_and_exceptions(**kwargs["input_kwargs"])
+        overrides += [
+            "data.do_include_subject_id=true",
+            "data.do_include_prediction_time=true",
+            "model.num_samples=2",
+            "data.max_seq_len=10",
+            "model.max_seq_len=20",
```

Comment:
This test passes if I set these to higher numbers, like 256 and 512, so I suspect an indexing bug related to the `global_end`, as it should be used to slice the JNRT prior to imposing the max_seq_len constraint.

Reply:
Does this pass now?